libs/matdbg/include/matdbg/DebugServer.h

```diff
@@ -99,7 +99,10 @@ class DebugServer {
     const backend::Backend mBackend;
 
     CivetServer* mServer;
+
     tsl::robin_map<MaterialKey, MaterialRecord> mMaterialRecords;
+    mutable utils::Mutex mMaterialRecordsMutex;
```

Comment:
Do we need to hold the lock inside `~DebugServer`?

Reply:
Yes, thanks.  I also rearranged it a little so that the server terminates before free-ing the records.